docs/framework/react/guide/search-params.md

```diff
@@ -172,6 +172,147 @@ In the above example, we used Zod's `.catch()` modifier instead of `.default()`
 
 The underlying mechanics why this works relies on the `validateSearch` function throwing an error. If an error is thrown, the route's `onError` option will be triggered (and `error.routerCode` will be set to `VALIDATE_SEARCH` and the `errorComponent` will be rendered instead of the route's `component` where you can handle the search param error however you'd like.
 
+#### Adapters
+
+When using a library like [Zod](https://zod.dev/) to validate search params you might want to `transform` search params before committing the search params to the URL. A common `zod` `transform` is `default` for example.
+
+```typescript
```

Comment:
Can we decorate these markdown code blocks as just `tsx` like the rest of the page? so we don't introduce much divergence in the docs.